### PR TITLE
[WIP] Sass upgrade starter

### PR DESCRIPTION
**Pull request by neoeno**
After chatting with @benbarnett I got excited and rebased/updated the branches I used to spike out the sass upgrade a few months ago. 

This branch points all the relevant engine gems to github branches called `unpin_sass_rails`. Those branches unpin `sass-rails`, and where necessary bump/remove other gems (e.g. `bootstrap-sass`).

I also got curious and ran `rake assets:clobber assets:precompile` on frontend to see what happened, and ended up fixing one of the issues before I managed to force myself to stop. There's still plenty of `@extend`s to be fixed.

Feel free to rebase/modify/commit to this branch. Same goes for the branches on the engines.

## Gem Progress

- [ ] Advice plans
- [x] Budget planner
- [ ] Car cost tool
- [ ] Debt advice locator
- [x] Debt free day calculator
- [ ] Decision trees
- [ ] Dough
- [ ] Mortgage calculator
- [ ] Pensions calculator
- [ ] Savings calculator
- [ ] Yeast

### CODE REVIEW

**Comment by benbarnett** [comment]
This is super helpful, thanks @neoeno!

**Comment by neoeno** [comment]
:warning: Got curious again. :warning: This will now pass `rake assets:clobber assets:precompile` if you do the following:

1. Checkout #master.
2. Apply [this patch](https://gist.github.com/neoeno/d280a2c4fbfd98c1bad9) to use the yeast spike branch.
3. Run `bundle && bundle exec bowndler install`.
4. Switch to `#fix_sass_rails_breakages`.
5. Run `bundle`. NOT `bowndler` — I forget, but there's a quirk of bowndler that messes up with git sourced gems.
6. Run `sed -ie s/is-errored/form__row--is-errored/ vendor/assets/bower_components/dough/assets/stylesheets/components/common/_validation_summary.scss`. This applies the changes to dough manually, since I can't work out how to get bowndler to use the github repo.
7. Run `rake assets:clobber assets:precompile`.

Obviously, when the engine/dough/yeast branches are merged into master all of that will become unnecessary. At that point we'll be able to run further tests on the result.

Some of the changes here and in the branches are speculative and could do with review from the maintainers. However, I'd be happy to start raising PRs on the respective repositories — they should all be fine to merge without this, but they will require some review. I've got a big load of questions I want to ask @benbarnett about things disappearing ;)

File: Gemfile [hunk 1]
```diff
@@ -40,18 +40,18 @@ gem 'rubytree'
 gem 'statsd-ruby'
 gem 'uglifier', '>= 1.3.0'
 
-gem 'advice_plans', '~> 3.1.0'
+gem 'advice_plans', git: 'git@github.com:moneyadviceservice/advice_plans.git', branch: 'unpin_sass_rails'
```
Comment (benlovell) [inline Gemfile]: You can golf these down slightly with `github: 'moneyadviceservice/repo' #...` thus cutting out the boilerplate git stuff.

Comment (neoeno) [inline Gemfile]: Yeah, I was thinking that when I looked through it... no idea why I didn't. Anyway, luckily this bit won't end up getting merged.

**Comment by darokel** [comment]
Can I check the status of this PR? @neoeno

**Comment by benbarnett** [comment]
It's part of a much bigger thing, we need to take it on as a group really.

**Comment by neoeno** [comment]
This and its linked PRs are complete, officially 'pending review', but I doubt they're going anywhere immediately, for the reason @benbarnett pointed out.

If this isn't useful though, we should close em all. @benbarnett?

**Pull request closed without merging**